ts/lib/commands/precompile.ts

```diff
@@ -0,0 +1,72 @@
+import os from 'os';
+import execa from 'execa';
+import fs from 'fs-extra';
+import path from 'path';
+import { command } from '../utilities/ember-cli-entities';
+import copyDeclarations from '../utilities/copy-declarations';
+
+export const PRECOMPILE_MANIFEST = 'dist/.ts-precompile-manifest';
+
+export default command({
+  name: 'ts:precompile',
+  works: 'insideProject',
+  description:
+    'Generates JS and declaration files from TypeScript sources in preparation for publishing.',
+
+  availableOptions: [{ name: 'manifest-path', type: String, default: PRECOMPILE_MANIFEST }],
+
+  async run(options: { manifestPath: string }) {
+    let outDir = `${os.tmpdir()}/e-c-ts-precompile-${process.pid}`;
+    let { paths, rootDir, pathRoots } = this._loadConfig(outDir);
+    if (!paths) {
+      this.ui.writeLine('No `paths` were found in your `tsconfig.json`, so `ts:precompile` is a no-op.');
+      return;
+    }
+
+    // prettier-ignore
+    await execa('tsc', [
+      '--declarationDir', outDir,
+      '--rootDir', rootDir || this.project.root,
+      '--allowJs', 'false',
+      '--noEmit', 'false',
+      '--declaration',
+      '--emitDeclarationOnly',
+      '--sourceMap', 'false',
+      '--inlineSourceMap', 'false',
+      '--inlineSources', 'false',
+    ]);
```

Comment:
I'm not sure what the reasoning is behind hardcoding these options, rather than using what's found the appropriate tsconfig (if one is found), overriding `--emitDeclarationOnly`. 

**A use case**  - if I have strictness settings defined, and `noEmitOnError` true, I'd expect no declaration files to be generated unless the strict type-checking works.

We also should consider `--declarationMap` (TS 2.9+)

Reply:
This was carried over from [the previous implementation](https://github.com/typed-ember/ember-cli-typescript/pull/388/files#diff-f11fe69778211644d8597f7e5f089f58L27) (the file has just changed enough over the commits here that it's treated as a delete/create in the full diff).

The flags here are layered on _over_ what's in `tsconfig.json`, the same as they would be when invoking `tsc` by hand, so the scenario you're describing already works exactly as you'd expect. In fact, several of these flags are only necessary in case their opposite is set in `tsconfig.json` (e.g. `tsc` complains if you have both `--declaration` and `--allowJs`, so we have to explicitly turn the latter off).

Reply:
I do agree that investigating the new declaration sourcemaps stuff is a good idea, though it may be that we can just allow users to configure that in their `tsconfig.json` and not need to set it explicitly. Now that we're not dealing with `.ts` -> `.js` transpilation here, we can actually likely drop the sourcemap related flags entirely and let the user do what they like. I'll update that file accordingly.